### PR TITLE
Add urban dictionary extension

### PR DESCRIPTION
**Pull request by naman1-gupta**
## Description
An extension to search [urban dictionary](https://urbandictionary.com) for a term

## Screencast

https://user-images.githubusercontent.com/23036464/154352819-825beb76-158e-4168-9430-2d011617c4a9.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/urban-dictionary/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "urban-dictionary",
+  "title": "Urban Dictionary",
+  "description": "Search urban dictionary for a term",
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/package.json:5]:
```suggestion
  "description": "Search urban dictionary for a word",
```

File: extensions/urban-dictionary/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        </List>
+    )
+}
+
+const searchUrbanDictionary = async (term?: string):Promise<Term[]> => {
+    
+    var options: AxiosRequestConfig = {
+       method: 'GET',
+       url: `http://api.urbandictionary.com/v0/define?term=${term}`,
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/src/index.tsx:80]:
```suggestion
       url: `https://api.urbandictionary.com/v0/define?term=${term}`,
```

File: extensions/urban-dictionary/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    useEffect(() => {
+        async function fetchTerms() {
+          try {
+            if(query) {
+                setIsLoading(true)
+                let res: Term[] = []
+                res = await searchUrbanDictionary(query)
+                setIsLoading(false)
+                setTerms(res);
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/src/index.tsx:29]:
```suggestion
                setTerms(res);
                setIsLoading(false)
```

File: extensions/urban-dictionary/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            style: Toast.Style.Failure,
+        })
+    }
+
+
+    return (
+        <List 
+            onSearchTextChange={(text) => setQuery(text)}
+            throttle
+            searchBarPlaceholder='Enter term to search UD...'
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/src/index.tsx:52]:
```suggestion
            searchBarPlaceholder='Enter word to search...'
```

File: extensions/urban-dictionary/package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  "icon": "icon.png",
+  "author": "naman",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Urban Dictionary",
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/package.json:12]:
```suggestion
      "title": "Search Word",
      "subtitle": "Urban Dictionary",
```

File: extensions/urban-dictionary/package.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      "description": "Search Urban Dictionary",
```
Comment (pernielsentikaer) [inline extensions/urban-dictionary/package.json:13]:
```suggestion
      "description": "Search Urban Dictionary a word",
```

File: extensions/urban-dictionary/src/index.tsx (hunk 2 above)
Comment (pernielsentikaer) [inline extensions/urban-dictionary/src/index.tsx:80]:
```suggestion
       url: `https://api.urbandictionary.com/v0/define?term=${term}`,
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @naman1-gupta 👋 

Awesome! Thanks for taking your time to contribute 🔥 

I left some suggestions and feedback for you to address:

* No fallback if the items like author (used in the list) – seems like there is more.
* Please run `npm run lint` and fix the errors mentioned
* Seems like you're using an older version of NPM

```
💥 error - validate package-lock.json file
wrong 'lockfileVersion' in package-lock.json. Found version 1, expected: 2. Update npm to v7.x
Error: wrong 'lockfileVersion' in package-lock.json. Found version 1, expected: 2. Update npm to v7.x
Error: Process completed with exit code 1.
```

Looking forward for the next version 🔥 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by naman1-gupta** [comment]
Hi @pernielsentikaer, 

Thank you for spending the time to review my code.

> - No fallback if the items like author (used in the list) – seems like there is more.
> - Please run npm run lint and fix the errors mentioned
> - Seems like you're using an older version of NPM

I've implemented a fix to all of these issues along with a few other suggestions. I don't agree with replacing `term` with `word` particularly because the search query could be a phrase as well and not necessarily a single word.

What do you think?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @naman1-gupta 👋 

The update is doing the job as expected, only reason I wanted to change it is because that's what urban dictionary call it – I'll leave it up to you with this one :)

dqLVCTJjvZEKR9ZF

Request a new review when you are ready (whatever you decide), feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by naman1-gupta** [comment]
> Hi @naman1-gupta 👋
> 
> The update is doing the job as expected, only reason I wanted to change it is because that's what urban dictionary call it – I'll leave it up to you with this one :)
> 
> dqLVCTJjvZEKR9ZF
> 
> Request a new review when you are ready (whatever you decide), feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Makes sense @pernielsentikaer . I've changed it to word to be consistent with the website.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @naman1-gupta 👋 

Awesome! 💪 One last correction that would improve the UX a bit is to return an empty array when you clear the search – right now it's possible to have a list of result and nothing in the search bar

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by fe9lix** [comment]
Hey guys, I'm closing this PR because of inactivity. Feel free to reopen it again later and address the feedback.

**Pull request closed without merging**